src/Audio.md

```diff
@@ -0,0 +1,103 @@
+
+## Common concepts
+
+### APU
+
+The Game Boy's sound chip is called the <abbr title="Audio Processing Unit">APU</abbr>.
+
+The APU runs off the same master clock as the rest of the Game Boy, which is to say, it is fully synced with the CPU and [PPU](<#Rendering Overview>).
+This also means that the APU runs about 2.4% faster on the SGB1, increasing frequencies by as much and thus sounding slightly higher-pitched.
+The SGB2 does not suffer from this issue.
+
+The APU processes a lot of events at a fixed rate, which depends on the type of event.
+Thus, the APU interfaces use durations instead of frequencies, which may be confusing as signal theory and music are more typically based on the latter.
+Thus, durations will be expressed from their frequencies: for example, a "256 Hz tick" means "1/256th of a second".
+
+The length of ticks is not affected by [CGB double speed](<#FF4D - KEY1 - CGB Mode Only - Prepare Speed Switch>), so the APU works just the same regardless of CPU speed.
+
+::: warning
+
+The Game Boy's APU is actually full of tricky details; this chapter will mostly describe the intended / common behavior, and often paper over bugs & quirks.
+Readers wishing to learn more should read the [APU details](<#APU details>) chapter.
+
+:::
+
+### Envelope
+
+Each channel's [**envelope**](https://en.wikipedia.org/wiki/Envelope_(music)) can be configured.
+A signal's [envelope](https://en.wikipedia.org/wiki/Envelope_(waves)) is, roughly speaking, the variation of the signal's amplitude.
+On the GB, "envelope" basically designates the channel's loudness.
+
+All channels can have their envelope controlled to modulate their volume (CH3 less precisely than other channels).
+In addition, all channels but CH3 benefit from a functionality called **envelope sweep**, which automatically adjusts the envelope over time.
+The parameters that can be controlled are how many times the envelope will be stepped, in which direction, and how long between steps.
+Envelope sweep ticks at 64 Hz, and every 1&ndash;7 of those ticks, the envelope will be increased or decreased.
+
+Unfortunately, of [ADSR envelopes](https://en.wikipedia.org/wiki/ADSR_envelope), the Game Boy only provides for Attack and Sustain, via envelope sweep and sound length respectively.
+
+### Sound length
+
+All channels can be set to automatically shut themselves down after a certain amount of time.
+This can be used to time the "sustain" part of ADSR, though the sound will cut fairly abruptly once the sound length "expires", instead of decaying.
+
+A channel's sound length is initially set between 0 and 63, and, if enabled, ticks up[^len_cnt_dir] at 256 Hz (tied to [DIV-APU](<#DIV-APU>)).
+When the counter reaches 64, the channel is turned off.
+
+[^len_cnt_dir]:
+Actually, the sound length is inverted when written, and *that* ticks down until it reaches 0.
+But the effect is as if the counter ticked up.
+
+### Wave length
```

Comment:
Ok with contracting.

What's the difference between the "conversion formula" and the "underlying function"?

Calling it "frequency" is outright wrong, since it's not what's being controlled, and has proven to cause confusion. This is why I'm changing it.

Reply:
By conversion formula, I mean literally the equation "Frequency = 131072/(2048-x) Hz" or some variant of it. 

By underlying function, I mean (a description of) how and why the oscillator functions the way it does.

And I'm not suggesting that the value itself should be referred to as a frequency, since it's obviously not. What I'm saying is that the section should possibly be called frequency, and things should be explained in terms of controlling the frequency. This makes educational sense because it's describing the hardware starting form familiar territory. (Assuming you have knowledge about music/sound theory.)

As a first approximation, this could be described as a special pitch value used by the hardware which can be calculated from the frequency.

Then a later paragraph can describe the exact meaning of the value, and how it's used internally by the hardware.

Reply:
Resolved locally by using "Frequency" as the section name. The conversion formula is not provided in the overview, simply because it depends on the channel.